### ISSUE TITLE
ComplEx model loss always the same

### ISSUE THREAD

**Issue by akalino** [thunlp/OpenKE#114]
When I run the PyTorch implementation of ComplEx, I see that my loss is the same for each batch. I see that he batching is working, the scoring is working, yet the softplus returns the same gradient no matter what the input. Any thoughts as to why this is happening? I do not have similar issues with other models. 

Example:
Epoch 0 | loss: 69.314722
Epoch 1 | loss: 69.314723
Epoch 2 | loss: 69.314720
Epoch 3 | loss: 69.314720
Epoch 4 | loss: 69.314717

**Comment by THUCSTHanxu13** [comment]
Maybe you use SGD to train models. For ComplEx, Adagrad is required to reduce its loss function.

**Comment by yangapku** [comment]
Sorry, it seems the error still occurs when Adagrad is used.

**Comment by Chrixtar** [comment]
I am having the same problem.
Does anyone know how to fix this?

**Comment by akalino** [comment]
When I swapped to Adagrad I did not face the issue anymore. Perhaps make
sure you delete any compiled versions of the code with the prior optimizer
still specified?

**Comment by Chrixtar** [comment]
Thanks, I found my issue:
I set the weight_decay for the optimizer, which does not seem to work with the SoftPlus function somehow....
